tests/test_environments/test_skills_evaluator_fallback.py

```
@@ -0,0 +1,101 @@
+            env._score("def f(): return 1", "reference solution", {})
+        assert isinstance(excinfo.value.__cause__, ValueError)
+
+    def test_evaluator_failure_falls_back_with_optin(self):
+        env = _assembled("code", _RaisingEvaluator(), allow_fallback=True)
+        reward, details = env._score("def f(): return 1", "reference solution", {})
+        assert reward == 0.0
+        assert details["method"] == "exact_match_fallback"
+
+    def test_working_evaluator_unaffected(self):
+        class _Working:
+            def eval_single(self, sample):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `eval_single`

(ANN202)